authorize_request_handler.go

```diff
@@ -210,6 +210,23 @@ func (f *Fosite) validateResponseTypes(r *http.Request, request *AuthorizeReques
 	request.ResponseTypes = responseTypes
 	return nil
 }
+func (f *Fosite) validateResponseMode(r *http.Request, request *AuthorizeRequest) error {
```

Comment:
This validation function should also respect security considerations from [here](https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Security):

> There are security implications to encoding response values in the query string. The HTTP Referer header includes query parameters, and so any values encoded in query parameters will leak to third parties. Thus, while it is safe to encode an Authorization Code as a query parameter when using a Confidential Client (because it can't be used without the Client Secret, which third parties won't have), more sensitive information such as Access Tokens and ID Tokens MUST NOT be encoded in the query string. In no case should a set of Authorization Response parameters whose default Response Mode is the fragment encoding be encoded using the query encoding.

Also see [description of combinations](https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Combinations).

Based on that it is not really true that all combinations are possible. For example, if default allows fragment, then query cannot be used. Moreover, for errors, it seems that you can use always whichever the client requested.

So this is really that if by default query string is used, you can request that fragment is used. But not the opposite.

Reply:
Given that for error any mode can be used, then maybe checking for valid combinations cannot be done here but in write handler?

Reply:
Added the check for using `query` while the default mode is `fragment` and returns an error in the `autheorize_response_writer.go`.